### PR TITLE
TM_CTED-2922: updating default region, auto add block

### PR DESCRIPTION
**Pull request by tmathebula2ucom**

### CODE REVIEW

File: course.php [hunk 1]
```diff
@@ -30,10 +32,27 @@
...
 
+$block_manager = $PAGE->blocks;
+$block_manager->add_region('content');
+$block_manager->load_blocks(true);
+
+// Check if block instance does not yet exist
+$gs_exists = false;
+foreach ($block_manager->get_regions() as $region) {
+    foreach ($block_manager->get_blocks_for_region($region) as $block) {
+        if ($block->instance->blockname == "side_bar") {
```
Comment (brett-fritz) [inline course.php:47]: Should you not be checking for 'blocks-side_bar-course' ?

Comment (brett-fritz) [reply course.php:47]: What if a different sidebar block was present?  you cannot assume it will be the side bar course one right?

Comment (tmathebula2ucom) [reply course.php:47]: No, as that would be more of a page visibility name as in applicable_formats(). The 'side_bar' here is the DOM block loaded in the 'content' region of this current page only.

Comment (tmathebula2ucom) [reply course.php:47]: Well I expect it to be in that region - I am only limiting the scope of my check to the 'content' region. So if it exists elsewhere, then we continue to add it there.

File: course.php [hunk 2]
```diff
@@ -30,10 +32,27 @@
...
+
 // Check user is logged in and capable of grading.
 require_login($course, false);
 
+$block_manager = $PAGE->blocks;
+$block_manager->add_region('content');
+$block_manager->load_blocks(true);
+
+// Check if block instance does not yet exist
+$gs_exists = false;
```
Comment (BrendonPretorius) [inline course.php]: Please follow Moodle naming convention

**Review by BrendonPretorius — CHANGES_REQUESTED** [review]

File: course.php (hunk 2 above)
Comment (tmathebula2ucom) [reply course.php]: Done

**Review by BrendonPretorius — APPROVED** [review]

**Review by loyisongq — APPROVED** [review]

**Pull request merged**